lib/spack/spack/oci/opener.py

```diff
@@ -382,6 +382,12 @@ def credentials_from_mirrors(
 def create_opener():
     """Create an opener that can handle OCI authentication."""
     opener = urllib.request.OpenerDirector()
+
+    proxies = urllib.request.getproxies()
+    if proxies:
+        proxy_handler = urllib.request.ProxyHandler(proxies)
```

Comment:
maybe shorter to just add it below in the list, that's what urllib seems to do https://github.com/python/cpython/blob/71aecc284efdf997939568a4167dbffe1a65b9bf/Lib/urllib/request.py#L547C38-L547C52

```python
[
urllib.request.ProxyHandler(),
urllib.request.UnknownHandler(),
...
]
```

Reply:
Isn't `build_opener` actually exactly what we want?

Reply:
I believe the point was to explicitly omit `HTTPHandler`, so we cannot accidentally pass credentials over HTTP upon unsafe redirect.

Reply:
I see. `urllib.request.ProxyHandler()` it is then.